### ISSUE TITLE
Flask-Migrate not triggering/emitting SQLAlchemy "after_create" event after instantiated

### ISSUE THREAD

**Issue by caffeinatedMike** [miguelgrinberg/Flask-Migrate#344]
I'm using an app-factory setup with `Flask-Migrate`. In my models file I have a few functions with a `SQLAlchemy.event.listens_for` decorator  in order to populate some tables with predefined, static entries after the tables are created. However, I've found that when using `Flask-Migrate`, these functions are not triggered. The tables are created, but since the functions aren't triggered they are all empty.

Steps I'm doing to run the app:
```
flask db init
flask db migrate -m "initial commit"
flask db upgrade
flask start
```

Below is my code:

## top-level .flaskenv
```
FLASK_ENV=development
FLASK_APP=start_dev.py
```

## top-level start_dev.py
```python3
from app import create_app, db

# Running "flask run" from command-line starts the application
app = create_app()

# Allows us to import db from "flask shell"
@app.shell_context_processor
def make_shell_context():
    return {'db': db}
```

## top-level config.py
```python3
from dotenv import load_dotenv
import os

basedir = os.path.abspath(os.path.dirname(__file__))
load_dotenv(os.path.join(basedir, '.env'))

class BaseConfig(object):
    SQLALCHEMY_TRACK_MODIFICATIONS = False

class TestConfig(BaseConfig):
    SECRET_KEY = 'shhhhh'
    ENV = 'development'
    SQLALCHEMY_DATABASE_URI = 'sqlite:///' + os.path.join(basedir, 'app.db')
```

## app/__init__.py
```python3
from flask_sqlalchemy import SQLAlchemy
from flask_migrate import Migrate
from flask import Flask

db = SQLAlchemy()
migrate = Migrate()

def create_app(config_obj='config.TestConfig'):
    app = Flask(__name__)
    app.config.from_object(config_obj)
    
    db.init_app(app)
    # This doesn't trigger sqlalchemy.events
    # I thought Migrate() was supposed to take care of the db.create_all()?
    migrate.init_app(app, db)
    """
    # This works
    with app.app_context():
        from . import models
        db.create_all()
    """
    
    from .frontend import bp as frontend_bp
    app.register_blueprint(frontend_bp)
    
    from .api import bp as api_bp
    app.register_blueprint(api_bp, url_prefix='/api')
    
    return app
```

## app/models.py
```python3
from sqlalchemy import event
from app import db

class Project(db.Model):
    __tablename__ = 'project'
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(150), index=True)
    is_template = db.Column(db.Boolean, default=False)
    
@event.listens_for(Project.__table__, 'after_create')
def create_test_project(*args, **kwargs):
    db.session.add(Project(name='Test Project'))
    db.session.commit()
```

Am I doing something wrong or did I misunderstand how `Flask-Migrate` was intended to be used? I'm happy to continue using the `app.app_context()` "with" block in conjunction with the `migrate.init_app(app)` line, but that could potentially cause some issues for people.

**Example Scenario when keeping the "with" block:**

- Someone develops their application utilizing the above "with" block in conjunction with `migrate.init_app(app)`
- They create all their db models and do the original migration  using the `flask db ...` commands on the command line.
- They later need to add a new table that other tables are going to be referencing.
- They create that table, and add the reference fields to the existing tables.
- **They forget to run the `flask db upgrade` commands before deploying the new code.**
- A short time later they realize a serious bug with the new code and need to rollback the changes. Uh oh! There's no migration record that knows to remove the relationships from the existing tables! Now they're in a pickle.

**TLDR;** I guess what I'm trying to ask is is there a way to make sure that `migrate.init_app()`:
1. Triggers `sqlalchemy.events`
2. Auto-records changes to existing models and creations of new models? I realize that this behavior might only be needed in a production environment to avoid potential snafus like the above, so maybe the extension could check for a config value such as `MIGRATE_AUTOCOMMIT_CHANGES`. That way the setting could be set appropriately in the different environments, like so:
## top-level (modified) config.py
```python3
from dotenv import load_dotenv
import os

basedir = os.path.abspath(os.path.dirname(__file__))
load_dotenv(os.path.join(basedir, '.env'))

class BaseConfig(object):
    SQLALCHEMY_TRACK_MODIFICATIONS = False

class TestConfig(BaseConfig):
    SECRET_KEY = 'shhhhh'
    ENV = 'development'
    SQLALCHEMY_DATABASE_URI = 'sqlite:///' + os.path.join(basedir, 'app.db')
    MIGRATE_AUTOCOMMIT_CHANGES = False

class LiveConfig(BaseConfig):
    SECRET_KEY = 'shhhhh'
    ENV = 'production'
    SQLALCHEMY_DATABASE_URI = os.environ.get('DATABASE_URL')
    MIGRATE_AUTOCOMMIT_CHANGES = True
```

**Comment by miguelgrinberg** [comment]
You have a wrong idea of what the role of Flask-Migrate is. This extension helps with the database migrations, nothing more. In fact, this extension does absolutely nothing while you run your application server. It only works when you run a `flask db ...` command.

You have to decide if you want to maintain your database tables through database migrations (i.e. Flask-Migrate) or through `db.create_all()`. You cannot use both at the same time as they compete with each other. The equivalent Flask-Migrate command to `db.create_all()` is `flask db upgrade`, which you would run from the CLI before running your server.

**Comment by caffeinatedMike** [comment]
@miguelgrinberg Thank you for clarifying. I guess my question now is would it be _possible_ to add some additional functionality to Flask-Migrate (the two points mentioned above), which _would_ run when the extension was instantiated inside the app-factory? I like using Flask-Migrate over `db.create_all()` due to its record-keeping nature and, in turn, the ability to rollback any changes when needed.

**Comment by miguelgrinberg** [comment]
@caffeinatedMike No, that does not really go well with the way this extension works. There are many functions, not just one thing. And not everyone wants to upgrade the database to the latest release every time they run the application, sometimes you need to work on an older version of the database. Build a compound command in your OS that combines `flask db upgrade` and `flask run` if you want to do these two together.

**Comment by caffeinatedMike** [comment]
Understood, thanks again for the clarification and insight!

Best Regards,

Michael Hill
